content/docs/cosidriver/features/objectscale.md

```diff
@@ -0,0 +1,327 @@
+metadata:
+  name: my-bucketaccessclass
+driverName: cosi.dellemc.com
+authenticationType: Key
+parameters:
+  id: "my.objectscale"
+```
+
+#### `authenticationType`
+
+> ⚠ **WARNING:** this field is case sensitive, and the granting access will fail if it is not set exactly to *Key* or *IAM*.
+
+`authenticationType` denotes the style of authentication. The only supported option for COSI Driver is `Key`.
+
+#### Unsupported options
+
+- `authenticationType=IAM` - denotes the style of authentication. The `IAM` value MUST NOT be used, because IAM style authentication is not supported.
+
+### Bucket Access
+
+`BucketAccess` resource represents a access request to generate a `Secret`, that will allow you to access ObjectStorage . The following is a sample manifest for creating a BucketClaim resource:
+
+```yaml
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketAccess
+metadata:
+  name: my-bucketaccess
+  namespace: my-namespace
+spec:
+  bucketClaimName: my-bucketclaim
+  protocol: S3
+  bucketAccessClassName: my-bucketaccessclass
+  credentialsSecretName: my-s3-secret
+```
+
+#### `spec.protocol`
+
+> ⚠ **WARNING:** this field is case sensitive, and the provisioning will fail if protocol is not set exactly to *S3*.
+
+`spec.protocol` is the name of the Protocol that this access credential is supposed to support.
+
+#### Unsupported options
+
+- `spec.serviceAccountName=...` - is the name of the serviceAccount that COSI will map to the object storage provider service account when IAM styled authentication is specified. As the IAM style authentication is not supported, this field is also unsupported.
+- `spec.protocol=...` - Protocols are the set of data API this bucket is required to support. From protocols specified by COSI (`v1alpha1`), Dell ObjectScale platform only supports the `S3` protocol. Protocols `Azure` and `GCS` MUST NOT be used.
+
+## Provisioning Buckets
+
+### Kubernetes Administrator Steps
+
+The first and foremost step to do before you can start provisioning object storage, is to create a `BucketClass`. The `BucketClass` is an object that defines the provisioning and management characteristics of `Bucket` resources. It acts as an abstraction layer between users (such as applications or pods) and the underlying object storage infrastructure. `BucketClass` allows you to dynamically provision and manage `Buckets` in a consistent and automated manner.
+
+The provided code snippet demonstrates how to create a `BucketClass`.
+
+```sh
+cat <<EOF | kubectl create --filename -
+apiVersion: objectstorage.k8s.io/v1alpha1
+kind: BucketClass
+metadata:
+  name: my-bucketclass
+driverName: cosi.dellemc.com
+deletionPolicy: Delete
+parameters:
+  id: "my.objectscale"
+EOF
+```
+
+### End-user Steps
+
+#### Greenfield Provisioning
+
+_Greenfield Provisioning_ means creating a new bucket from scratch, without any existing data.
+
+The provided code snippet demonstrates how to create a `BucketClaim` for greenfield provisioning.
```

Comment:
The following example shows how to create a 'BucketClaim' for greenfield provisioning.